src/realty/domain/realty.spec.ts

```diff
@@ -1,86 +1,145 @@
+          title: 'some title',
+          description: validDescription,
+        });
+        expect(realty.description).toBe(validDescription);
+      });
+    });
+    describe('when null description', () => {
+      it('should set description null when description is null', () => {
+        const nullDescription = null;
+        const realty = new Realty({
+          title: 'some title',
+          description: nullDescription,
+        });
+        expect(realty.description).toBe(null);
+      });
+    });
+    describe('when invalid description', () => {
+      const EXPECTED_ERROR =
+        'description must be less than or equal to 500 characters';
+      it('should throw error when description length is more than 500 characters', () => {
+        const longDescription =
+          'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Phasellus nec elit et arcu malesuada porttitor vitae eleifend augue. Proin efficitur elit nunc, fermentum gravida elit eleifend ut. Donec bibendum maximus risus, quis aliquet nisl fermentum at. Nulla volutpat venenatis orci non venenatis. Sed at ligula vel dui faucibus auctor ac in nulla. Nam ut augue quis urna sodales vulputate. Maecenas efficitur, neque nec pulvinar volutpat, erat turpis blandit tellus, at malesuada diam felis eu tortor..';
+        expect(
+          () =>
+            new Realty({
+              title: 'some title',
+              description: longDescription,
+            }),
+        ).toThrowError(EXPECTED_ERROR);
+      });
     });
-    expect(realty.description).toBe(null);
   });
+  describe('setPrice', () => {
+    describe('when valid price', () => {
+      it('should set price when price is valid', () => {
+        const validPrice = 150000;
+        const realty = new Realty({
+          title: 'some title',
+          price: validPrice,
+        });
+        expect(realty.price).toBe(validPrice);
+      });
+    });
 
-  it('should throw an error when price is less than 0', () => {
-    const invalidPrice = -100;
-    expect(() =>
-      Realty.create({
-        title: 'some title',
-        price: invalidPrice,
-      }),
-    ).toThrowError('price must be greater than 0');
-  });
+    describe('when null price', () => {
+      it('should set price null when price is null', () => {
+        const nullPrice = null;
+        const realty = new Realty({
+          title: 'some title',
+          price: nullPrice,
+        });
+        expect(realty.price).toBe(nullPrice);
+      });
+    });
 
-  it('should throw an error when price is 0', () => {
-    const invalidPrice = 0;
-    expect(() =>
-      Realty.create({
-        title: 'some title',
-        price: invalidPrice,
-      }),
-    ).toThrowError('price must be greater than 0');
-  });
+    describe('when invalid price', () => {
+      const EXPECTED_ERROR = 'price must be a positive number';
+      it('should throw error when price is less than 0', () => {
+        const invalidPrice = -100;
```

Comment:
Nessa organização, acredito que todos os `when` deveriam estar em um `describe` e nunca em um `it`. Daí talvez fique com níveis demais. Eu preferiria remover o `when invalid price` e deixar o contexto mais fechado com `when price is less than 0`, `when price is 0`, etc...

```suggestion
    describe('when invalid price', () => {
      const EXPECTED_ERROR = 'price must be a positive number';
      
      describe('when price is less than 0', () => {
          it('should throw error', () => {
              const invalidPrice = -100;
```
```suggestion
      describe('when price is less than 0', () => {
          it('should throw error', () => {
              const EXPECTED_ERROR = 'price must be a positive number';
              const invalidPrice = -100;
```

Quanto à variável de erro, pode parecer ruim assim, mas tenho outra sugestão sobre isso e te passo em outro comentário.